### DISCUSSION TITLE
Can BPN validation be bypassed?

### DISCUSSION THREAD

**Discussion by SebastianOpriel** [eclipse-tractusx/tractusx-edc#521] [Q&A] [answered]
## Describe the bug

I would like to know if [BPN validation](https://github.com/eclipse-tractusx/tractusx-edc/blob/377a6d01813b5ee1c4c9af48972f5ec347e1d357/edc-extensions/business-partner-validation/src/main/java/org/eclipse/tractusx/edc/validation/businesspartner/functions/AbstractBusinessPartnerValidation.java#L77) can be bypassed if Catena-X portal does not check Connector URLs of onboarded connectors (need to admit, not being too deep into BPN details yet). Thus, how this risk is mitigated currently?

### Potential vector
1. You will need a BPN of a connector, which you want to mimic referred to "BPN_OF_INTEREST".
2. Create a connector for URL https://{{BPN_OF_INTEREST}}.yourdomain.com/[...]/OWN_BPN
4. Register connector in CX portal
5. Try to negotiate a contract with a data offering which has BPN validation active

### Expected behavior
Not being able to negotiate a contract

## Possible Implementation to mitigate risk
use `endswith` instead of `contains`

**Comment by paullatzelsperger** [comment, marked as answer]
FYI I converted this to a discussion. Also, the BPN validation received a significant overhaul with the introduction of SSI and VerifiableCredentials in version `0.5.x`.

**Comment by jimmarino** [comment]
In the old release, someone would have to circumvent portal security and register a false connector. 

With release `0.5.x`, this is not possible unless someone obtains the connector`s summary credential.

**Comment by stephanbcbauer** [comment]
@SebastianOpriel did the answers help? Were you already able to test it again?
